Client/Frontend/Browser/TabManager.swift

```diff
@@ -453,14 +453,40 @@ class TabManager: NSObject {
                     for delegate in self.delegates {
                         delegate.get()?.tabManagerDidAddTabs(self)
                     }
+                } else {
+                    self.showFocusPromoToast()
                 }
                 self.eraseUndoCache()
             })
         }
 
         delegates.forEach { $0.get()?.tabManagerDidRemoveAllTabs(self, toast: toast) }
     }
-    
+
+    func showFocusPromoToast() {
+        if !LeanplumIntegration.sharedInstance.shouldShowFocusUI() {
+            return
+        }
+
+        guard let templateDict = LeanplumIntegration.sharedInstance.getTemplateDictionary(), let labelText = templateDict["Template Text"], let buttonText = templateDict["Button Text"], let colorText = templateDict["Hex Color String"], let deepLink = templateDict["Deep Link"] else {
+            return
+        }
+
+        LeanplumIntegration.sharedInstance.track(eventName: .focusPromoImpression, withParameters: ["button name": buttonText as AnyObject])
+        let toast = LeanplumPromoButtonToast(labelText: labelText, buttonText: buttonText, colorText: colorText, completion: { buttonPressed in
+            guard let url = URL(string: deepLink) else {
+                return
+            }
+
+            if UIApplication.shared.canOpenURL(url) && buttonPressed {
```

Comment:
That is a great point @jhugman - if we send down a bad url from LP's dashboard, we shouldn't show the toast if the user can't even act on it. Of course we would test this so it shouldn't be a _practical_ concern, but James is right to prevent it technically.